src/test/util_tests.cpp

```diff
@@ -42,6 +42,17 @@ namespace BCLog {
 
 BOOST_FIXTURE_TEST_SUITE(util_tests, BasicTestingSetup)
 
+BOOST_AUTO_TEST_CASE(util_datadir)
+{
+    ClearDatadirCache();
+    const fs::path dd_norm = GetDataDir();
+    {
+        gArgs.ForceSetArg("-datadir", dd_norm.string() + "/");
```

Comment:
nit, could add another case lile `+ "/./"` to it's clear the result is canonical, not just without last `/`.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/20080#issuecomment-704794437).